### PR TITLE
Update default TLD from .dev to .test ... because Google.

### PR DESCRIPTION
**Pull request by aloyr**
Change default TLD from "dev" to "test" because of HSTS for .dev on Chrome 63
    
tl;dr:
Google owns .dev and made changes to Chrome that enforces HSTS to .dev domains, which breaks local environments reliant on that TLD.
    
This change swaps .dev for .test as default dev domains, as specified on RFC2606.
    
Info about the change in Chrome:
https://ma.ttias.be/chrome-force-dev-domains-https-via-preloaded-hsts/
https://chromium-review.googlesource.com/c/chromium/src/+/669396/10/net/http/transport_security_state_static.json
    
Info about .dev:
https://icannwiki.org/.dev
    
Info about .test:
https://en.wikipedia.org/wiki/.test
https://tools.ietf.org/html/rfc2606

Additionally, this also swaps "_" for "-" to reduce collision of default docker names
    
Default Docker container names are composed of <adjective>_<person_name>, this causes increased naming collisions, as do container names from docker-compose and docker stack.
    
This change swaps "_" for "-" on container names, allowing a more natural solution for container names.

### CODE REVIEW

**Comment by nlighteneddesign** [comment]
LGTM

**Comment by katannshaw** [comment]
Same here @aloyr . Looks good.

**Comment by chrisoconnor001** [comment]
LGTM too :)

**Review by ruudud — APPROVED** [review]
Thanks for a solid contribution! Looks good.

**Comment by ruudud** [comment]
I'm a bit unsure about the conversion of `_` to `-` though, because it breaks the behaviour mentioned [here](https://github.com/ruudud/devdns/blob/master/README.md#container-name-to-dns-record-conversion). I also recognise that it's a bit confusing given the Docker naming scheme, so I'll look in to it.

I'll for sure pull in the TLD change though.

**Comment by ruudud** [comment]
Pulled in the `.dev` → `.test` in  77c5569. I'll look into making the `-` conversion optional.

**Comment by aloyr** [comment]
Thanks @ruudud.

One of the reasons I worked on the `-` conversion is that when using `docker-compose`, each container is prefixed with the name of the stack and `_`. 

Here is an example of what I am talking about:
```bash
docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
484c8d015815        aloyr/devdns        "/run.sh"                3 weeks ago         Up 31 hours         0.0.0.0:53->53/udp       docker_dns_1
ad539aa64e16        nginx               "nginx -g 'daemon of…"   3 weeks ago         Up 31 hours         0.0.0.0:80->80/tcp       docker_www_1
d0e638547b3e        aloyr/php           "docker-php-entrypoi…"   3 weeks ago         Up 31 hours         9000/tcp                 docker_php_1
30aa2f8fd89a        mariadb             "docker-entrypoint.s…"   3 weeks ago         Up 31 hours         0.0.0.0:3306->3306/tcp   docker_db_1
1bb64ef0c4ad        redis:alpine        "docker-entrypoint.s…"   3 weeks ago         Up 31 hours         0.0.0.0:6379->6379/tcp   docker_redis_1
```

Another reason that motivated me is that [underscores are not allowed in URLs in the hostname as per RFC 952, assumptions, section 1](https://tools.ietf.org/html/rfc952).

I hope the above served to clarify some points, and thanks again for the merge. :)
